Classes/Plugin/Tools/SearchInDocumentTool.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace Kitodo\Dlf\Plugin\Tools;
+
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+use Kitodo\Dlf\Common\Helper;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * SearchInDocument tool for the plugin 'Toolbox' of the 'dlf' extension
+ *
+ * @author Sebastian Meyer <sebastian.meyer@slub-dresden.de>
+ * @author Alexander Bigga <alexander.bigga@slub-dresden.de>
+ * @package TYPO3
+ * @subpackage dlf
+ * @access public
+ */
+class SearchInDocumentTool extends \Kitodo\Dlf\Common\AbstractPlugin {
+    public $scriptRelPath = 'Classes/Plugin/Tools/SearchInDocumentTool.php';
+
+    /**
+     * The main method of the PlugIn
+     *
+     * @access public
+     *
+     * @param string $content: The PlugIn content
+     * @param array $conf: The PlugIn configuration
+     *
+     * @return string The content that is displayed on the website
+     */
+    public function main($content, $conf) {
+
+        $this->init($conf);
+
+        // Merge configuration with conf array of toolbox.
+        if (!empty($this->cObj->data['conf'])) {
+            $this->conf = Helper::mergeRecursiveWithOverrule($this->cObj->data['conf'], $this->conf);
+        }
+
+        $this->addSearchInDocumentJS();
+
+        // Load current document.
+        $this->loadDocument();
+        if ($this->doc === NULL
+            || $this->doc->numPages < 1
+            || empty($this->conf['fileGrpFulltext'])
+            || empty($this->conf['solrcore'])) {
+            // Quit without doing anything if required variables are not set.
+            return $content;
+        }
+
+        // Quit if no fulltext file is present
+        $fullTextFile = $this->doc->physicalStructureInfo[$this->doc->physicalStructure[$this->piVars['page']]]['files'][$this->conf['fileGrpFulltext']];
+        if (empty($fullTextFile)) {
+            return $content;
+        }
+
+        // Load template file.
+        $this->getTemplate();
```

Comment:
Template loading for tools has changed slightly (see #392).

```suggestion
        $this->getTemplate('###TEMPLATE###', '', TRUE);
```